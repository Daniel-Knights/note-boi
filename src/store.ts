import { reactive } from 'vue';
import { invoke } from '@tauri-apps/api/tauri';
import { v4 as uuidv4 } from 'uuid';

import { isEmptyNote } from './utils';

export class Note {
  readonly id = uuidv4();
  body = {
    delta: '',
    text: '',
  };
  timestamp = Date.now();
}

interface State {
  notes: Note[];
  selectedNote: Note;
}

export const state = reactive<State>({
  notes: [],
  selectedNote: new Note(),
});

/** Sorts notes in descending order by timestamp. */
function sortNotes() {
  state.notes.sort((a, b) => b.timestamp - a.timestamp);
}

/** Finds note index within {@link state.notes}. */
function findNoteIndex(id: string) {
  return state.notes.findIndex((nt) => nt.id === id);
}

/** Finds note within {@link state.notes}. */
export function findNote(id: string): Note | undefined {
  return state.notes.find((nt) => nt.id === id);
}

/** Deletes {@link state.selectedNote} when both `title` and `body` are empty. */
function clearEmptyNote(): void {
  const isValidClear = state.notes.length > 1;

  const foundNote = findNote(state.selectedNote.id);
  if (!foundNote) return;

  if (isValidClear && isEmptyNote(foundNote)) {
    deleteNote(state.selectedNote.id);
  }
}

const selectEvent = new CustomEvent('note-select');

/**
 * Looks for note with given `id` in {@link state.notes},
 * and sets it to {@link state.selectedNote}.
 */
export function selectNote(id: string): void {
  if (state.selectedNote.id === id) return;

  clearEmptyNote();

  const foundNote = findNote(id);
  if (foundNote) state.selectedNote = { ...foundNote };

  document.dispatchEvent(selectEvent);
}

/** Fetches all notes and updates {@link state}. */
export async function getAllNotes(): Promise<void> {
  const fetchedNotes = await invoke<Note[]>('get_all_notes').catch(console.error);

  const hasNotes = fetchedNotes && fetchedNotes.length > 0;
  if (!hasNotes) return newNote();

  state.notes = fetchedNotes;

  sortNotes();

  if (isEmptyNote(fetchedNotes[0])) {
    state.notes[0].timestamp = Date.now();
  }

  state.selectedNote = { ...state.notes[0] };

  document.dispatchEvent(selectEvent);
}

/** Deletes note with the given `id`. */
export function deleteNote(id: string): void {
  state.notes.splice(findNoteIndex(id), 1);
  state.selectedNote = { ...state.notes[0] };

  sortNotes();

  invoke('delete_note', { id }).catch(console.error);
}

/** Creates an empty note. */
export function newNote(): void {
  const foundNote = findNote(state.selectedNote.id);

  // Only update timestamp if selected note is empty
  if (foundNote && isEmptyNote(foundNote)) {
    state.selectedNote.timestamp = Date.now();
    return;
  }

  const freshNote = new Note();

  state.notes.unshift(freshNote);
  state.selectedNote = { ...freshNote };

  invoke('new_note', { note: { ...freshNote } }).catch(console.error);

  document.dispatchEvent(selectEvent);
}

/** Edits note body on Quill `text-change`. */
export function editBody(delta: string, text: string): void {
  const foundNote = findNote(state.selectedNote.id);
  if (!foundNote || text === foundNote.body.text) return;

  const timestamp = Date.now();
  foundNote.timestamp = timestamp;
  state.selectedNote.timestamp = timestamp;

  foundNote.body = { delta, text };

  sortNotes();

  invoke('edit_note', { note: { ...foundNote } }).catch(console.error);
}
