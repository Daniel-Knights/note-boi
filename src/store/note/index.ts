export * from './event';
export * from './operations';
export * from './state';
export * from './types';
export * from './utils';
