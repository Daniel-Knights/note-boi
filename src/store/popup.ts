import { ref } from 'vue';

export enum PopupType {
  Auth,
  Error,
  Info,
  ChangePassword,
}

export const openedPopup = ref<PopupType>();
