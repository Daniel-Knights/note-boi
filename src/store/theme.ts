import { ref } from 'vue';

import { STORAGE_KEYS } from '../constant';

export const COLOUR_THEMES = ['Light', 'Dark', 'System'] as const;

type ColourTheme = (typeof COLOUR_THEMES)[number];

export const selectedTheme = ref<ColourTheme>(
  (localStorage.getItem(STORAGE_KEYS.THEME) as ColourTheme | null) || 'System'
);

export function setTheme(theme: ColourTheme): void {
  document.body.classList.remove(`theme--${selectedTheme.value.toLowerCase()}`);
  document.body.classList.add(`theme--${theme.toLowerCase()}`);

  localStorage.setItem(STORAGE_KEYS.THEME, theme);

  selectedTheme.value = theme;
}

document.body.classList.add(`theme--${selectedTheme.value.toLowerCase()}`);
