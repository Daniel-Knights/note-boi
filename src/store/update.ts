import * as dialog from '@tauri-apps/plugin-dialog';
import { relaunch } from '@tauri-apps/plugin-process';
import { check, Update } from '@tauri-apps/plugin-updater';
import { ref } from 'vue';

export const update = ref<Update>();
export const updateDownloading = ref<boolean>(false);

export async function updateAndRelaunch(): Promise<void> {
  if (!update.value) return;

  updateDownloading.value = true;

  try {
    await update.value?.downloadAndInstall();

    update.value = undefined;

    await relaunch();
  } catch (error) {
    console.error(error);

    const shouldRetry = await dialog.ask('Try again?', {
      title: 'Unable to install update',
      kind: 'error',
    });

    if (shouldRetry) {
      await updateAndRelaunch();
    } else {
      update.value = undefined;
      updateDownloading.value = false;
    }
  }
}

// TODO: swap these functions
export async function handleUpdate(): Promise<void> {
  const checkResult = await check();
  if (!checkResult?.available) return;

  update.value = checkResult;

  const newVersion = checkResult.version;

  // Check if the user has already been notified
  const seenVersion = localStorage.getItem('update-seen');
  if (seenVersion === newVersion) {
    update.value = undefined;

    return;
  }

  const shouldInstall = await dialog.ask(
    'A new version of NoteBoi is available.\nDo you want to update now?',
    `Update available: v${newVersion}`
  );
  if (!shouldInstall) {
    localStorage.setItem('update-seen', newVersion);

    update.value = undefined;

    return;
  }

  updateAndRelaunch();
}
