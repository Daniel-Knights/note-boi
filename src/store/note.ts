import * as dialog from '@tauri-apps/plugin-dialog';
import type Delta from 'quill-delta';
import { reactive } from 'vue';

import { NOTE_EVENTS } from '../constant';
import { isEmptyNote, tauriInvoke } from '../utils';

import { autoPush, syncState, UnsyncedNoteIds } from './sync';

export type UnsyncedEventDetail = {
  noteId: string;
  kind: keyof Pick<UnsyncedNoteIds, 'edited' | 'deleted'>;
};

export class Note {
  readonly id: string = crypto.randomUUID();
  timestamp = Date.now();
  content = {
    delta: <Partial<Delta>>{
      ops: [],
    },
    title: '',
    body: '',
  };
}

export const noteState = reactive({
  notes: <Note[]>[],
  selectedNote: new Note(),
  extraSelectedNotes: <Note[]>[],
});

const newNoteEvent = new CustomEvent(NOTE_EVENTS.new);
export const selectNoteEvent = new CustomEvent(NOTE_EVENTS.select);
export const changeNoteEvent = new CustomEvent(NOTE_EVENTS.change);

function getUnsyncedEvent(
  noteId: UnsyncedEventDetail['noteId'],
  kind: UnsyncedEventDetail['kind']
) {
  return new CustomEvent(NOTE_EVENTS.unsynced, { detail: { noteId, kind } });
}

/** Sorts notes in descending order by timestamp. */
export const sortNotesFn = (a: Note, b: Note): number => b.timestamp - a.timestamp;

/** Sorts {@link noteState.notes} in descending order by timestamp. */
export const sortStateNotes = (): Note[] => noteState.notes.sort(sortNotesFn);

/** Finds note index within {@link noteState.notes}. */
export function findNoteIndex(id?: string): number {
  return noteState.notes.findIndex((nt) => nt.id === id);
}

/** Finds note within {@link noteState.notes}. */
export function findNote(id?: string): Note | undefined {
  return noteState.notes.find((nt) => nt.id === id);
}

/** Deletes {@link noteState.selectedNote} when note is empty. */
function clearEmptyNote(): void {
  const isValidClear = noteState.notes.length > 1;
  if (!isValidClear) return;

  const foundNote = findNote(noteState.selectedNote.id);
  if (!foundNote) return;

  if (isEmptyNote(foundNote)) {
    deleteNote(noteState.selectedNote.id);
  }
}

/**
 * Looks for note with given `id` in {@link noteState.notes},
 * and sets it to {@link noteState.selectedNote}.
 */
export function selectNote(id?: string): void {
  if (noteState.selectedNote.id === id) return;

  clearEmptyNote();

  const foundNote = findNote(id);
  if (!foundNote) return;

  noteState.selectedNote = { ...foundNote };

  document.dispatchEvent(selectNoteEvent);
  document.dispatchEvent(changeNoteEvent);
}

/**
 * Returns true if note is either {@link noteState.selectedNote}
 * or within {@link noteState.extraSelectedNotes}.
 */
export function isSelectedNote(note: Note): boolean {
  return (
    note.id === noteState.selectedNote.id ||
    noteState.extraSelectedNotes.some((nt) => nt?.id === note.id)
  );
}

/** Fetches all notes and updates {@link noteState}. */
export async function getAllNotes(): Promise<void> {
  const fetchedNotes = await tauriInvoke('get_all_notes').catch(console.error);

  const hasNotes = fetchedNotes && fetchedNotes.length > 0;
  if (!hasNotes) return newNote();

  noteState.notes = fetchedNotes;

  sortStateNotes();

  const isSingleEmptyNote = fetchedNotes.length === 1 && isEmptyNote(fetchedNotes[0]);

  if (isSingleEmptyNote) {
    noteState.notes[0]!.timestamp = Date.now();
    // Clear these fields as whitespace-only can affect empty note checks
    noteState.notes[0]!.content.delta = {};
    noteState.notes[0]!.content.title = '';
    noteState.notes[0]!.content.body = '';
  }

  noteState.selectedNote = { ...noteState.notes[0]! };

  if (!isSingleEmptyNote) {
    clearEmptyNote();

    document.dispatchEvent(changeNoteEvent);
  }
}

/** Deletes note with the given `id`. */
export function deleteNote(id: string): void {
  noteState.notes.splice(findNoteIndex(id), 1);

  if (noteState.notes.length === 0) {
    newNote();
  } else if (noteState.selectedNote.id === id) {
    noteState.selectedNote = { ...noteState.notes[0]! };

    document.dispatchEvent(selectNoteEvent);
    document.dispatchEvent(changeNoteEvent);
  }

  if (syncState.unsyncedNoteIds.new === id) {
    syncState.unsyncedNoteIds.add({ new: '' });
  } else {
    document.dispatchEvent(getUnsyncedEvent(id, 'deleted'));

    tauriInvoke('delete_note', { id })
      .then(() => autoPush())
      .catch(console.error);
  }
}

/** Deletes {@link noteState.selectedNote} and all notes in {@link noteState.extraSelectedNotes}. */
export function deleteSelectedNotes(): void {
  deleteNote(noteState.selectedNote.id);

  noteState.extraSelectedNotes.forEach((nt) => {
    deleteNote(nt.id);
  });

  noteState.extraSelectedNotes = [];
}

/** Creates an empty note. */
export function newNote(isButtonClick?: boolean): void {
  const foundNote = findNote(noteState.selectedNote.id);

  // Only update timestamp if selected note is empty
  if (foundNote && isEmptyNote(foundNote)) {
    noteState.selectedNote.timestamp = Date.now();

    return;
  }

  const freshNote = new Note();

  noteState.notes.unshift(freshNote);
  noteState.selectedNote = { ...freshNote };

  // Ensure fresh note isn't overwritten on pull
  if (isButtonClick) {
    syncState.unsyncedNoteIds.add({ new: freshNote.id });
  }

  document.dispatchEvent(selectNoteEvent);
  document.dispatchEvent(changeNoteEvent);
  document.dispatchEvent(newNoteEvent);

  tauriInvoke('new_note', { note: { ...freshNote } }).catch(console.error);
}

/**
 * Edits currently selected note on Quill `text-change`.
 *
 * Selected note content should only be edited within `noteState.notes`, not
 * `noteState.selectedNote`, as `noteState.selectedNote` is what's reflected in the
 * editor. The timestamp, however, should be updated for both.
 */
export function editNote(delta: Partial<Delta>, title: string, body?: string): void {
  const foundNote = findNote(noteState.selectedNote.id);
  if (!foundNote || delta === foundNote.content.delta) return;

  const timestamp = Date.now();

  foundNote.timestamp = timestamp;
  noteState.selectedNote.timestamp = timestamp;

  foundNote.content = { delta, title, body: body || '' };

  sortStateNotes();

  document.dispatchEvent(getUnsyncedEvent(foundNote.id, 'edited'));

  tauriInvoke('edit_note', { note: { ...foundNote } })
    .then(autoPush)
    .catch(console.error);
}

/** Exports all notes, or a given selection. */
export async function exportNotes(noteIds: string[]): Promise<void> {
  const saveDir = await dialog.open({
    title: 'Choose a location',
    directory: true,
    multiple: false,
    recursive: false,
  });
  if (!saveDir) return;

  const notes = noteState.notes.filter((nt) => noteIds?.includes(nt.id));

  tauriInvoke('export_notes', { saveDir, notes }).catch(console.error);
}
