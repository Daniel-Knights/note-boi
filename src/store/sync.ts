import { reactive } from 'vue';
import { http, invoke } from '@tauri-apps/api';
import type { Response } from '@tauri-apps/api/http';

import { getAllNotes, Note, state as noteState } from './note';
import { isDev, tauriEmit } from '../utils';

export enum ErrorType {
  None,
  Auth,
  Push,
  Pull,
  Logout,
}

interface State {
  username: string;
  password: string;
  token: string;
  hasUnsyncedNotes: boolean;
  isLoading: boolean;
  isLogin: boolean; // For switching login/signup form
  autoSyncEnabled: boolean;
  error: {
    type: ErrorType;
    message: string;
  };
}

const autoSync = localStorage.getItem('auto-sync') as 'true' | 'false' | null;

export const state = reactive<State>({
  username: localStorage.getItem('username') || '',
  password: '',
  token: localStorage.getItem('token') || '',
  hasUnsyncedNotes: false,
  isLoading: false,
  isLogin: true,
  autoSyncEnabled: autoSync !== null ? autoSync === 'true' : true,
  error: {
    type: ErrorType.None,
    message: '',
  },
});

// Utils //

/** Parses an error response and returns a formatted message. */
function parseErrorRes(res: Response<Record<string, unknown>>) {
  const unknownErrorMessage = 'Unknown error, please try again';

  if (!res.data) return unknownErrorMessage;

  return typeof res.data.error === 'string' ? res.data.error : unknownErrorMessage;
}

/** Resets {@link state.error}. */
export function resetError(): void {
  state.error = { type: ErrorType.None, message: '' };
}

/** Wrapper for {@link http.fetch}. */
function tauriFetch<T>(
  endpoint: string,
  method: 'POST' | 'PUT',
  payload?: Record<string, unknown>
): Promise<Response<T>> {
  const baseUrl = isDev() ? 'http://localhost:8000' : 'TODO';

  return http.fetch<T>(`${baseUrl}/api${endpoint}`, {
    method,
    body: {
      type: 'Json',
      payload,
    },
  });
}

function clientSideLogout() {
  state.token = '';
  state.username = '';

  localStorage.removeItem('token');
  localStorage.removeItem('username');
}

/** Sets user preference for auto-syncing to {@link state} and `localStorage` */
export function setAutoSync(enabled: boolean): void {
  state.autoSyncEnabled = enabled;
  localStorage.setItem('auto-sync', enabled ? 'true' : 'false');
}

// Routes //

export async function login(): Promise<void> {
  state.isLoading = true;

  const res = await tauriFetch<Record<string, string | Note[]>>('/login', 'POST', {
    username: state.username,
    password: state.password,
  });

  if (res.ok) {
    resetError();
    tauriEmit('login');

    await invoke('sync_all_local_notes', { notes: res.data.notes }).catch(console.error);
    await getAllNotes().catch(console.error);

    state.token = res.data.token as string;
    state.password = '';

    localStorage.setItem('username', state.username);
    localStorage.setItem('token', state.token);
  }

  state.isLoading = false;

  if (!res.ok) {
    state.error = {
      type: ErrorType.Auth,
      message: parseErrorRes(res),
    };

    console.error(res.data);
  }
}

export async function signup(): Promise<void> {
  state.isLoading = true;

  const res = await tauriFetch<Record<string, string>>('/signup', 'POST', {
    username: state.username,
    password: state.password,
    notes: noteState.notes,
  });

  state.isLoading = false;

  if (res.ok) {
    resetError();
    tauriEmit('login');

    state.token = res.data.token;
    state.password = '';

    localStorage.setItem('username', state.username);
    localStorage.setItem('token', state.token);
  } else {
    state.error = {
      type: ErrorType.Auth,
      message: parseErrorRes(res),
    };

    console.error(res.data);
  }
}

export async function logout(): Promise<void> {
  if (!state.token) return; // Prevent bug where event.emit triggers event.listen

  state.isLoading = true;

  const res = await tauriFetch<Record<string, never | string>>('/logout', 'POST', {
    username: state.username,
    token: state.token,
  });

  state.isLoading = false;

  if (res.ok) {
    resetError();
    tauriEmit('logout');
    clientSideLogout();
  } else {
    state.error = {
      type: ErrorType.Logout,
      message: parseErrorRes(res),
    };
  }
}

export async function pull(): Promise<void> {
  state.isLoading = true;

  const res = await tauriFetch<Record<string, string | Note[]>>('/notes', 'POST', {
    username: state.username,
    token: state.token,
  });

  if (res.ok) {
    resetError();
    await invoke('sync_all_local_notes', { notes: res.data.notes }).catch(console.error);
    await getAllNotes().catch(console.error);
  }

  state.isLoading = false;

  if (!res.ok) {
    state.error = {
      type: ErrorType.Pull,
      message: parseErrorRes(res),
    };

    // User not found
    if (res.status === 404) clientSideLogout();

    console.error(res.data);
  }
}

export async function push(): Promise<void> {
  state.isLoading = true;

  const res = await tauriFetch<Record<string, never | string>>('/notes', 'PUT', {
    username: state.username,
    token: state.token,
    notes: noteState.notes,
  });

  state.isLoading = false;

  if (res.ok) {
    resetError();
    state.hasUnsyncedNotes = false;
  } else {
    state.error = {
      type: ErrorType.Push,
      message: parseErrorRes(res),
    };

    console.error(res.data);
  }
}

document.addEventListener('note-unsynced', () => {
  if (state.token) {
    state.hasUnsyncedNotes = true;
  }
});
