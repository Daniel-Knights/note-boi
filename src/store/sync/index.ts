import { http } from '@tauri-apps/api';
import type { Response } from '@tauri-apps/api/http';
import { reactive } from 'vue';

import { STORAGE_KEYS } from '../../constant';
import { isDev } from '../../utils';
import { UnsyncedNoteIds, unsyncedNoteIds } from './note';

export enum ErrorType {
  None,
  Auth,
  Push,
  Pull,
  Logout,
}

export const syncState = reactive({
  username: localStorage.getItem(STORAGE_KEYS.USERNAME) || '',
  password: '',
  token: localStorage.getItem(STORAGE_KEYS.TOKEN) || '',
  isLoading: false,
  isLogin: true, // For switching login/signup form
  error: {
    type: ErrorType.None,
    message: '',
  },
  unsyncedNoteIds: <UnsyncedNoteIds>{
    new: unsyncedNoteIds.new || '',
    edited: new Set<string>(unsyncedNoteIds.edited),
    deleted: new Set<string>(unsyncedNoteIds.deleted),
    get size() {
      return this.edited.size + this.deleted.size;
    },
    clear() {
      this.new = '';
      this.edited.clear();
      this.deleted.clear();
      localStorage.removeItem(STORAGE_KEYS.UNSYNCED);
    },
    add(ids): void {
      if (ids.new !== undefined) this.new = ids.new;

      ids.edited?.forEach((id) => this.edited.add(id));

      // Only track deleted notes if logged in
      ids.deleted?.forEach((id) => {
        this.deleted.add(id);

        if (this.edited.has(id)) {
          this.edited.delete(id);
        }
      });

      if (this.edited.has(this.new) || this.deleted.has(this.new)) {
        this.new = '';
      }

      localStorage.setItem(
        STORAGE_KEYS.UNSYNCED,
        JSON.stringify({
          new: this.new,
          edited: [...this.edited],
          deleted: [...this.deleted],
        })
      );
    },
  },
});

// Utils //

/** Parses an error response and returns a formatted message. */
export function parseErrorRes(res: Response<Record<string, unknown>>): string {
  const unknownErrorMessage = 'Unknown error, please try again';

  if (!res.data) return unknownErrorMessage;

  return typeof res.data.error === 'string' ? res.data.error : unknownErrorMessage;
}

/** Resets {@link state.error}. */
export function resetError(): void {
  syncState.error = { type: ErrorType.None, message: '' };
}

/** Wrapper for {@link http.fetch}. */
export function tauriFetch<T>(
  endpoint: string,
  method: 'POST' | 'PUT',
  payload?: Record<string, unknown>
): Promise<Response<T>> {
  const baseUrl = isDev()
    ? 'http://localhost:8000'
    : 'https://note-boi-server.herokuapp.com';

  return http.fetch<T>(`${baseUrl}/api${endpoint}`, {
    method,
    body: {
      type: 'Json',
      payload,
    },
  });
}

/** Catches hanging requests (e.g. due to server error). */
export function catchHang(err: unknown, type: ErrorType): void {
  console.error(err);

  syncState.isLoading = false;
  syncState.error = {
    type,
    message: 'Request failed',
  };
}

export { autoPush, push, pull, syncNotes, UnsyncedNoteIds } from './note';
export { clientSideLogout, login, signup, logout } from './auth';
export { deleteAccount } from './account';
