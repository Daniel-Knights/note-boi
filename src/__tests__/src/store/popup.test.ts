import { openedPopup, PopupType } from '../../../store/popup';

describe('Popup', () => {
  it('Sets opened popup', () => {
    assert.isUndefined(openedPopup.value);
    openedPopup.value = PopupType.Auth;
    assert.strictEqual(openedPopup.value, PopupType.Auth);
    openedPopup.value = PopupType.Error;
    assert.strictEqual(openedPopup.value, PopupType.Error);
    openedPopup.value = PopupType.Info;
    assert.strictEqual(openedPopup.value, PopupType.Info);

    assert.lengthOf(Object.keys(PopupType), 4 * 2);
  });
});
