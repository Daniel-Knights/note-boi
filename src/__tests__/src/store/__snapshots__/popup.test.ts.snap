// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup > Sets opened popup > KeyStore.getKey() 1`] = `undefined`;

exports[`Popup > Sets opened popup > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`Popup > Sets opened popup > Storage.get('THEME') 1`] = `null`;

exports[`Popup > Sets opened popup > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`Popup > Sets opened popup > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`Popup > Sets opened popup > Storage.get('USERNAME') 1`] = `null`;

exports[`Popup > Sets opened popup > allCalls 1`] = `[]`;

exports[`Popup > Sets opened popup > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`Popup > Sets opened popup > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`Popup > Sets opened popup > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`Popup > Sets opened popup > openedPopup 1`] = `"info"`;

exports[`Popup > Sets opened popup > selectedTheme 1`] = `"System"`;

exports[`Popup > Sets opened popup > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`Popup > Sets opened popup > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;
